server/services/prisonerService.ts

```diff
@@ -56,4 +57,23 @@ export default class PrisonerService {
     }
     return new PrisonerSearchApiClient(token).searchPrisonersByNomsIds(prisonerSearchCriteria)
   }
+
+  async getHdcStatuses(username: string, offenders: Prisoner[]): Promise<HdcStatus[]> {
+    const listOfHdcStatus: HdcStatus[] = []
+    const token = await this.hmppsAuthClient.getSystemClientToken(username)
+    const client = new PrisonApiClient(token)
+    await Promise.all(
+      offenders.map(async o => {
+        try {
+          const hdc = await client.getLatestHdcStatus(o.bookingId)
```

Comment:
Just wonder if there's a limit to how many async threads node can handle at once, I dont know the answer

Reply:
Seems ok at 20-30. I don't like this either, and suggested to David and Jon that the caseload page is probably the wrong one, or to adjust the prison-API endpoint to a POST accepting a list of booking Ids. Its short-term, but working for now.

Reply:
The best solution would be an intermediate page after selection where the offenders/ licences are narrowed down to one.